### ISSUE TITLE
Unified project repo's with all-repos

### ISSUE THREAD

**Issue by foarsitter** [py-pdf/pypdf_table_extraction#3]
I would propose that we unify all the repositories under the py-pdf organization with [all-repos](https://github.com/asottile/all-repos). I would like to see that within the py-pdf organization projects the same tools and configurations are used where possible so the possibility arises to make adjustments cross-repo with `all-repos`. 

This gives us the benefits that all the maintainers now how to run the tests and do some maintenance tasks like updating a python version and extending the test-matrix because they are familiar with the project structure without knowing the code.

**Comment by bosd** [comment]
I agree, Ideally we would utilize the same toolset within al repo's.
I have not evaluated yet if that is possible.

On the other hand, it is good to get started here and have some mechanisims in place.
Which covers #2.
Some linters, pre-commit stuff, ruff, tests, autopublish to pypi.
@foarsitter I remember you had some ideas about this in the other repo.

**Comment by foarsitter** [comment]
@bosd I used the [cookiecutter-hypermodern-python](
  https://cookiecutter-hypermodern-python.readthedocs.io/en/2022.6.3.post1/guide.html) to scaffold the previous fork and merged back in the source. 

> Here is a detailed list of features for this Python template:
>
> - Packaging and dependency management with [Poetry](https://python-poetry.org/)
> - Test automation with [Nox](https://nox.thea.codes/)
> - Linting with [pre-commit](https://pre-commit.com/) and [Flake8](http://flake8.pycqa.org/)
> - Continuous integration with [GitHub Actions](https://github.com/features/actions)
> - Documentation with [Sphinx](http://www.sphinx-doc.org/), [MyST](https://myst-parser.readthedocs.io/), and [Read the Docs](https://readthedocs.org/) using the [furo](https://pradyunsg.me/furo/) theme
> - Automated uploads to [PyPI](https://pypi.org/) and [TestPyPI](https://test.pypi.org/)
> - Automated release notes with [Release Drafter](https://github.com/release-drafter/release-drafter)
> - Automated dependency updates with [Dependabot](https://dependabot.com/)
> - Code formatting with [Black](https://github.com/psf/black) and [Prettier](https://prettier.io/)
> - Import sorting with [isort](https://pycqa.github.io/isort/)
> - Testing with [pytest](https://docs.pytest.org/en/latest/)
> - Code coverage with [Coverage.py](https://coverage.readthedocs.io/)
> - Coverage reporting with [Codecov](https://codecov.io/)
> - Command-line interface with [Click](https://click.palletsprojects.com/)
> - Static type-checking with [mypy](http://mypy-lang.org/)
> - Runtime type-checking with [Typeguard](https://github.com/agronholm/typeguard)
> - Automated Python syntax upgrades with [pyupgrade](https://github.com/asottile/pyupgrade)
> - Security audit with [Bandit](https://github.com/PyCQA/bandit) and [Safety](https://github.com/pyupio/safety)
> - Check documentation examples with [xdoctest](https://github.com/Erotemic/xdoctest)
> - Generate API documentation with [autodoc](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) and [napoleon](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html)
> - Generate command-line reference with [sphinx-click](https://sphinx-click.readthedocs.io/)
> - Manage project labels with [GitHub Labeler](https://github.com/marketplace/actions/github-labeler)


The docs contain some tutorials: https://cookiecutter-hypermodern-python.readthedocs.io/en/2022.6.3.post1/guide.html#tutorials

**Comment by bosd** [comment]
@foarsitter  Yes, the hypermodern-cookiecutter. 
It looks very interesting, I researcged a bit about it after you referred to it over the the camelot repo.
Wanted to use it for another project, but did'nt get to implement it yet.

The cookiecutter project you're referring too, IMO still seems the best of the bunch.
Yet, It seems it could use some maintainer :heart:  as wel. (Last commit july 2023.)
(Still supports python 3.7, but nothing newer then 3.10)

We can stull use it as a base.  Merge the improvements which are worthwhile from one of the forks here.
